sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -508,4 +516,37 @@ final ColumnarMap getMap(int rowId) {
       super(vector);
     }
   }
+
+  private static class IntervalYearAccessor extends ArrowVectorAccessor {
+
+    private final IntervalYearVector accessor;
+
+    IntervalYearAccessor(IntervalYearVector vector) {
+      super(vector);
+      this.accessor = vector;
+    }
+
+    @Override
+    int getInt(int rowId) {
+      return accessor.get(rowId);
+    }
+  }
+
+  private static class IntervalDayAccessor extends ArrowVectorAccessor {
+
+    private final IntervalDayVector accessor;
+    private final NullableIntervalDayHolder intervalDayHolder = new NullableIntervalDayHolder();
+
+    IntervalDayAccessor(IntervalDayVector vector) {
+      super(vector);
+      this.accessor = vector;
+    }
+
+    @Override
+    long getLong(int rowId) {
+      accessor.get(rowId, intervalDayHolder);
+      return Math.addExact(intervalDayHolder.days * MICROS_PER_DAY,
```

Comment:
`days` has the `Int` type, `MICROS_PER_DAY` is Long but `intervalDayHolder.days * MICROS_PER_DAY` can overflow Long in general case. @Peng-Lei If you believe the overflow never happens, could proof that or/and add an assert.

Reply:
Yeah，I test it. It did overflow. Thank you very much.

Reply:
Such negative test could be useful, can you add it to the PR?

Reply:
OK, I'll try to add it

Reply:
done